internal/security/secretstore/kongadminapi.go

```diff
@@ -0,0 +1,250 @@
+	err = ioutil.WriteFile(k.paths.jwt, []byte(k.secrets.jwt.signed), 0400)
+	if err != nil {
+		return fmt.Errorf("%s Failed to write JWT to file %s: %w", k.prefixes.errText, k.paths.jwt, err)
+	}
+
+	return nil
+}
+
+// createJWT creates a JWT from a given private key and issuer payload value
+func (k *KongAdminAPI) createJWT(privateKey string, issuer string) (signedToken string, err error) {
+
+	// Setup JWT generation variables
+	now := time.Now().Unix()
+	duration, err := time.ParseDuration(k.jwtDuration)
+	if err != nil {
+		return "", fmt.Errorf("%s Could not parse JWT duration: %w", k.prefixes.errText, err)
+	}
+
+	// Sanity check - parse & check EC key
+	eckey, err := jwt.ParseECPrivateKeyFromPEM([]byte(privateKey))
+	if err != nil {
+		return "", fmt.Errorf("%s Could not parse private key: %w", k.prefixes.errText, err)
+	}
+	if eckey.Params().BitSize != 256 {
+		return "", fmt.Errorf("%s EC key bit size is incorrect (%d instead of 256)", k.prefixes.errText, eckey.Params().BitSize)
+	}
+
+	// Create JWT
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, &jwt.StandardClaims{
+		Issuer:    issuer,
+		IssuedAt:  now,
+		NotBefore: now,
+		ExpiresAt: now + int64(duration.Seconds()),
+	})
+
+	// Save JWT to string return
+	signedToken, err = token.SignedString(eckey)
+	if err != nil {
+		return "", fmt.Errorf("%s Could not sign JWT: %w", k.prefixes.errText, err)
+	}
+
+	return
+}
+
+// clearSecrets clears the key structs by setting them to empty struct values
+func (k *KongAdminAPI) clearSecrets() {
+	k.secrets = KongAdminAPISecrets{}
+}
+
+// verifyPrefixes confirms that there are values set for the prefixes
+func (k *KongAdminAPI) verifyPrefixes() error {
+
+	base := "[build-kong-admin]"
+
+	if k.prefixes.cmdText == "" {
+		k.prefixes.cmdText = base
+	}
+
+	if k.prefixes.errText == "" {
+		k.prefixes.errText = base + "[error]"
+	}
+
+	if k.prefixes.infoText == "" {
+		k.prefixes.infoText = base + "[info]"
+	}
+
+	return nil
+}
+
+// verifyPaths confirms values are set for the paths, and that those paths exist
+func (k *KongAdminAPI) verifyPaths() error {
+
+	// Is the template path set?
+	if k.paths.template == "" {
+		return fmt.Errorf("%s Template file path (<KongAdminAPI>.paths.template) is not set - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Does the template path/file exist? It should be set.
+	if !helper.CheckIfFileExists(k.paths.template) {
+		return fmt.Errorf("%s Template file path (<KongAdminAPI>.paths.template) does not exist - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Is the config file path set?
+	if k.paths.config == "" {
+		return fmt.Errorf("%s Kong configuration file path (<KongAdminAPI>.paths.config) is not set - check configuration.toml", k.prefixes.errText)
+	}
+
+	// Does the config file path directory exist? If not, create it.
+	if !helper.CheckIfFileExists(k.paths.config) {
+		dirErr := os.MkdirAll(path.Dir(k.paths.config), 0644)
```

Comment:
permission for directory should be `755` not `644`,

Reply:
Why? I feel like we changed it in a previous review from being more open to being more closed.